tools/Setup.csx

```diff
@@ -0,0 +1,228 @@
+                        break;
+                    case "vscode":
+                        Client = Client.VSCode;
+                        break;
+                    case "debug":
+                        Configuration = Configuration.Debug;
+                        break;
+                    case "release":
+                        Configuration = Configuration.Release;
+                        break;
+                    default:
+                        Console.WriteLine(string.Format("Unknown flag {0}", arg));
+                        break;
+                }
+            }
+            else
+            {
+                TargetPath = arg;
+            }
+        }
+
+        if (string.IsNullOrWhiteSpace(TargetPath))
+        {
+            throw new ArgumentNullException(nameof(TargetPath));
+        }
+    }
+
+    private class ListFileFormat
+    {
+        public string FileName { get; }
+        public string SourceRoot { get; }
+        public string SourceDir { get; }
+        public string InstallDir { get; }
+
+        public ListFileFormat(
+            string fileName,
+            string sourceRoot,
+            string sourceDir,
+            string installDir
+        )
+        {
+            FileName = fileName;
+            SourceRoot = sourceRoot;
+            SourceDir = sourceDir;
+            InstallDir = installDir;
+        }
+    }
+
+    private IList<ListFileFormat> ParseListFiles(string filePath)
+    {
+        List<ListFileFormat> fileFormats = new List<ListFileFormat>();
+
+        using (StreamReader fs = new StreamReader(filePath))
+        {
+            string line;
+
+            while((line = fs.ReadLine()) != null)  
+            {
+                line = line.Trim();
+
+                // Skip blank or commented out lines
+                if (string.IsNullOrWhiteSpace(line) || line.StartsWith("#"))
+                {
+                    continue;
+                }
+
+                string[] data = line.Split(',');
+                if (data.Length < 4)
+                {
+                    Console.Error.WriteLine(string.Format("Invalid line in {0}: '{1}'", filePath, line));
+                    continue;
+                }
+
+                ListFileFormat listFile = new ListFileFormat(
+                    data[0],
+                    data[1],
+                    data[2],
+                    data[3]
+                );
+                
+                fileFormats.Add(listFile);
+            }  
+        }
+
+        return fileFormats;
+    }
+
+    public void Run()
+    {
+        string scriptDirectoryPath = Path.GetDirectoryName(GetSourceFile());
+        string srcDirectoryPath = Path.GetFullPath(Path.Join(scriptDirectoryPath, "..", "src"));
+        string binDirectoryPath = Path.GetFullPath(Path.Join(scriptDirectoryPath, "..", "bin"));
+
+        // No client flag provided, try to auto detect.
+        if (Client == Client.None)
+        {
```

Comment:
Should this just be required to specify?

Reply:
I think auto resolving on target path is acceptable. Going to add automatically finding the cpptools extension path if they want VS Code.